NewForce-Cohort-6/tabloidfullstack-mac-cheese
# TS | Create category | #10

# Description

This code addresses ticket #10 (create category)

https://trello.com/b/0CrteZkv/mac-n-cheese

Given an admin is on the Category list page
When they select the Create Category button
Then they should be directed to a form in which they can enter a new category name

Given an admin has entered a Category name
When they click the Save button
Then a new category should be saved to the database
And the admin should be directed to the Category list page

NOTE: For the time being it is acceptable to treat all users as admin users. There is a future story about enforcing user permissions.

_____________________________________________________________
# Testing Instructions

* In your terminal, from your current branch:
-git add and git commit whatever you're working on in your branch
-git checkout main
-git fetch --a or git fetch all
-(if files do not show, git pull origin)
-git checkout CreateCategory-TS
-start the application with "start <application name>"

* Ensure that you have created the Tabloid database
* If needed:

-Use this to fill out tables and columns: 

https://github.com/NewForce-Cohort-6/tabloidfullstack-mac-cheese/blob/main/SQL/01_Tabloid_Create_DB.sql

-Use this to seed the database: 

https://github.com/NewForce-Cohort-6/tabloidfullstack-mac-cheese/blob/main/SQL/02_Tabloid_Seed_Data.sql

* Launch the api from VisualStudio using the filled in green 'play' button at the top
* Lauch the front end code by right clicking the 'client' folder in visual studio solution explorer and selecting 'Open in terminal'
* Once the terminal opens, type 'code .'
* This should open visual studio code. Open a new terminal and type 'npm start'
* In the browser, if not logged in already, register an email with the registration button, then log in with that email
* Once logged in, select the "Category Management" tab from the navbar
* Once the categories tab is selected, you should see a list of categories in alphabetical order with non-functional edit and delete buttons as well as a "New Category" button.
* Select the new category button, which should take you to a form to create a new category name.
* Type in a new category and select the save button, which should direct you back to the category list.
* Check the database from swagger to ensure the new category was created.

_____________
# Checklist:
